fe-build-image/action.yml

```
@@ -26,32 +27,92 @@ runs:
     - name: Set up Docker Buildx
       uses: docker/setup-buildx-action@v2
 
-    - name: Login to GitHub Container Registry
-      uses: docker/login-action@v2
+    - name: Set up Docker image name
+      id: image
+      shell: bash
+      run: |
+        echo "image_name=`basename ${{ github.repository }}`" >> $GITHUB_OUTPUT
+
+    - name: Set up Docker registries info
+      id: registries
+      shell: bash
+      run: |
+        IMAGE_NAME=${{ steps.image.outputs.image_name }}
+        echo "images<<EOF" >> $GITHUB_OUTPUT
+        echo "${{ inputs.registries }}" | while read line; do
+          echo "$line" | awk -F',' "{print \$1\"/\"\$2\"/$IMAGE_NAME\"}" >> $GITHUB_OUTPUT
+        done
+        echo "EOF" >> $GITHUB_OUTPUT
+
+    - name: Login to the custom registries
+      shell: bash
+      run: |
+        echo "${{ inputs.registries }}" | while read line; do
+          REGISTRY=$(echo $line | awk -F',' '{print $1}')
+          USERNAME=$(echo $line | awk -F',' '{print $3}')
+          PASSWORD=$(echo $line | awk -F',' '{print $4}')
+          echo "$PASSWORD" | docker login -u $USERNAME --password-stdin $REGISTRY
+        done
+
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v4
       with:
-        registry: ghcr.io
-        username: ${{ github.repository_owner }}
-        password: ${{ inputs.github_token }}
+        images: ${{ steps.registries.outputs.images }}
+        tags: |
+          type=ref,event=pr
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}},enable=${{ !startsWith(github.ref, 'refs/tags/v0.0.') }}
+          type=semver,pattern={{major}},enable=${{ !startsWith(github.ref, 'refs/tags/v0.') }}
+          type=sha
 
-    - name: Get APP Info
-      uses: byzanteam/jet-actions/fe-get-app-info@main
+    - if: ${{ inputs.cache_type == 'gha' }}
+      name: Build and push
+      uses: docker/build-push-action@v3
       with:
-        package_file: ${{ inputs.package_file }}
-      id: get-app-info
+        context: ${{ inputs.context }}
+        file: ${{ inputs.dockerfile }}
+        build-args: |
+          APP_VSN=${{ steps.meta.outputs.version }}
+          APP_NAME=${{ steps.image.outputs.image_name }}
+          SOURCE_REPO_URL=${{ github.repositoryUrl }}
+          SOURCE_SHA=${{ github.sha }}
+          ${{ inputs.build_args }}
+        push: true
+        labels: ${{ steps.meta.outputs.labels }}
+        tags: ${{ steps.meta.outputs.tags }}
+        cache-from: type=${{ inputs.cache_type }}
+        cache-to: type=${{ inputs.cache_type }},mode=max
```

这里不用再取变量了，if 已经判断了这里一定是 gha
```suggestion
        cache-from: type=gha
        cache-to: type=gha,mode=max
```